# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoader.h
# Commit: 0ac6d9d26e2b
# Full Hash: 0ac6d9d26e2ba97e31eb1f1a0e75d93071c2cf50
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 09:08:22
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - Split ScriptLoadRequest into ScriptLoadRequest and DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134041
# ==============================================================================

diff -r 5a00e56ade15 -r 0ac6d9d26e2b dom/script/ScriptLoader.h
--- a/dom/script/ScriptLoader.h	Thu Feb 17 15:45:19 2022 +0000
+++ b/dom/script/ScriptLoader.h	Thu Feb 17 15:57:48 2022 +0000
@@ -559,6 +559,12 @@
   // Implements https://html.spec.whatwg.org/#execute-the-script-block
   nsresult EvaluateScriptElement(ScriptLoadRequest* aRequest);
 
+  // Handles both offthread and synchronous modules
+  nsresult CompileOrFinishModuleScript(
+      JSContext* aCx, JS::Handle<JSObject*> aGlobal,
+      JS::CompileOptions& aOptions, ModuleLoadRequest* aRequest,
+      JS::MutableHandle<JSObject*> aModule) override;
+
   // Handles both bytecode and text source scripts; populates exec with a
   // compiled script
   nsresult CompileOrDecodeClassicScript(JSContext* aCx,
